dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImplTest.java

```diff
@@ -481,4 +487,145 @@ private void runLuceneQueryAndValidateResults(final String query, final Contentl
+                .languageId(language1.getId())
+                .setProperty("body", "myBody")
+                .nextPersisted();
+        
+        // create URL-Mapped content
+        final Contentlet publishedTwoLanguage2 = new ContentletDataGen(blogType.id())
+                .languageId(language2.getId())
+                .setProperty("body", "myBody")
+                .setProperty("identifier", workingTwoLanguage.getIdentifier())
+                .nextPersisted();
+        
+        
+        APILocator.getContentletAPI().publish(publishedTwoLanguage2, APILocator.systemUser(), false);
+        
+        
+        
+        assertEquals("workingOneLanguage exists and is working", workingOneLanguage, instance.findContentletByIdentifier(workingOneLanguage.getIdentifier(), false, language1.getId()));
+        assertNull("workingOneLanguage does not exist in 2nd language", instance.findContentletByIdentifier(workingOneLanguage.getIdentifier(), false, language2.getId()));
+        assertNull("workingOneLanguage does not exist in live", instance.findContentletByIdentifier(workingOneLanguage.getIdentifier(), true, language1.getId()));
+
+        assertNull("workingTwoLanguage in language1 is not live", instance.findContentletByIdentifier(workingTwoLanguage.getIdentifier(), true, language1.getId()));
+        assertEquals("workingTwoLanguage exists in langauge1 and is working", workingTwoLanguage, instance.findContentletByIdentifier(workingTwoLanguage.getIdentifier(), false, language1.getId()));
+        assertEquals("workingTwoLanguage exists in langauge2 and is working", publishedTwoLanguage2, instance.findContentletByIdentifier(workingTwoLanguage.getIdentifier(), false, language2.getId()));
+        assertEquals("workingTwoLanguage exists in langauge2 and is live", publishedTwoLanguage2, instance.findContentletByIdentifier(workingTwoLanguage.getIdentifier(), true, language2.getId()));
+
+    }
+    
+    /**
+     * This tests whether we are getting cached results from queries to elasticsearch and that these
+     * results are invalidated when a new piece of content is checked in
+     * 
+     * @throws Exception
+     */
+    @Test
+    public void test_cached_es_query_response() throws Exception {
+        
+        final Language language1 = new LanguageDataGen().nextPersisted();
+
+        final ContentType blogType = TestDataUtils.getBlogLikeContentType(site);
+        
+
+        
+        final String liveQuery = "+baseType:1 +live:true" ;
+        final String workingQuery = "+baseType:1 +live:false" ;
+        
+
+        SearchHits hits = instance.indexSearch(liveQuery, 10, 0, null);
+        
+        //assert we have results
+        assertTrue(hits.getTotalHits().value > 0);
+        
+        SearchHits hits2 = instance.indexSearch(liveQuery, 10, 0, null);
+        
+        // hits and hits2 are the same object in memory (meaning, it came from cache)
+        assertTrue(hits == hits2);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use assertSame(x, y) instead of assertTrue(x==y), or assertNotSame(x,y) vs assertFalse(x==y)](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5288908)